src/bin/client.rs

```diff
@@ -0,0 +1,56 @@
+use std::io::{self, Write};
+use std::time::Duration;
+use std::thread;
+use futures_util::{future, pin_mut, StreamExt};
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use tokio_tungstenite::{connect_async, tungstenite::protocol::Message};
+use clap::Parser;
+
+#[derive(Parser, Debug)]
+#[clap(version, about)]
+struct Args {
+    #[clap(short, long, default_value = "ws://localhost:8000/opcode")]
+    connect_addr: String,
+}
+
+#[tokio::main]
+async fn main() {
+    let args = Args::parse();
+
+    let url = url::Url::parse(&args.connect_addr).unwrap();
+    let (stdin_tx, stdin_rx) = futures_channel::mpsc::unbounded();
+    tokio::spawn(read_stdin(stdin_tx));
+
+    let (ws_stream, _) = connect_async(url).await.expect("Failed to connect");
+    println!("WebSocket handshake has been successfully completed");
+
+    let (write, read) = ws_stream.split();
+
+    let stdin_to_ws = stdin_rx.map(Ok).forward(write);
+    let ws_to_stdout = {
+        read.for_each(|message| async {
+            let data = message.unwrap().into_data();
```

Comment:
```suggestion
            let data = message.unwrap().into_data();
            tokio::io::stdout().write_all(&data).await.unwrap();
```

Fehler loggen und behandeln?